src/Sylius/Behat/Context/Domain/ManagingOrdersContext.php

```diff
@@ -147,6 +147,18 @@ public function thisOrderHasNotBeenPaidForDays(OrderInterface $order, $amount, $
         $this->unpaidOrdersStateUpdater->cancel();
     }
 
+    /**
+     * @Given /^the order with number "([^"]*)" has not been paid for (\d+) (day|days)$/
```

Comment:
If you change `order with number "([^"]*)"` to `(order "[^"]+")` and use `sylius.behat.context.transform.order` in the suite - this number would be automatically translated to Order entity :dancer: